tests/unit/package_managers/test_generic.py

```diff
@@ -0,0 +1,43 @@
+from unittest import mock
+
+import pytest
+
+from cachi2.core.errors import PackageRejected
+from cachi2.core.models.input import GenericPackageInput
+from cachi2.core.package_managers.generic import (
+    DEFAULT_LOCKFILE_NAME,
+    _resolve_generic_lockfile,
+    fetch_generic_source,
+)
+from cachi2.core.rooted_path import RootedPath
+
+
+@mock.patch("cachi2.core.package_managers.rpm.main.RequestOutput.from_obj_list")
+@mock.patch("cachi2.core.package_managers.generic._resolve_generic_lockfile")
+def test_fetch_generic_source(
+    mock_resolve_generic_lockfile: mock.Mock,
+    mock_from_obj_list: mock.Mock,
+) -> None:
+
+    mock_components = [mock.Mock()]
+    mock_resolve_generic_lockfile.return_value = mock_components
+
+    mock_request = mock.Mock()
+    model_input = GenericPackageInput.model_construct(type="generic")
+    mock_request.generic_packages = [model_input]
+
+    fetch_generic_source(mock_request)
+
+    mock_resolve_generic_lockfile.assert_called()
+    mock_from_obj_list.assert_called_with(components=mock_components)
+
+
+def test_resolve_generic_no_lockfile(rooted_tmp_path: RootedPath) -> None:
+    with pytest.raises(PackageRejected) as exc_info:
+        mock_source_dir = mock.Mock()
+        mock_source_dir.join_within_root.return_value.path.exists.return_value = False
+        _resolve_generic_lockfile(mock_source_dir, mock.Mock())
```

Comment:
That is unnecessary complicated.
You can simplify that for example like this:

```
    with pytest.raises(PackageRejected) as exc_info:
        _resolve_generic_lockfile(rooted_tmp_path, rooted_tmp_path)
```

Reply:
Thanks, fixed